tests/test_opds_import.py

```diff
@@ -2013,36 +2072,51 @@ def test_resources_are_mirrored_on_import(self):
             importer.import_from_feed(self.content_server_mini_feed,
                                       feed_url='http://root')
         )
-        e1 = imported_editions[0]
-        e2 = imported_editions[1]
 
         assert 2 == len(pools)
 
-        # The import process requested each remote resource in the
-        # order they appeared in the OPDS feed. The thumbnail
-        # image was not requested, since we never trust foreign thumbnails.
-        assert http.requests == [
-            'http://www.gutenberg.org/ebooks/10441.epub.images',
-            'https://s3.amazonaws.com/book-covers.nypl.org/Gutenberg-Illustrated/10441/cover_10441_9.png',
-            'http://www.gutenberg.org/ebooks/10557.epub.images',
-            'http://root/broken-cover-image',
-            'http://root/working-cover-image'
+        # Both items were requested
+        assert epub10441['url'] in http.requests
+        assert epub10557['url'] in http.requests
+
+        # The import process requested each remote resource in the feed. The thumbnail
+        # image was not requested, since we never trust foreign thumbnails. The order they
+        # are requested in is not deterministic, but after requesting the epub the images
+        # should be requested.
+        index = http.requests.index(epub10441['url'])
```

Comment:
There are a lot of changes to this test because during an OPDS import we iterate [over a dictionary of metadata values](https://github.com/NYPL-Simplified/server_core/blob/develop/opds_import.py#L779), that means that the order of requests we see in these tests can change depending on the order of that iteration. So we have to take into account that we don't know which book from the test feed will be imported first.

Reply:
This is fine for now but how about sorting that dictionary to guarantee a reliable order?

Reply:
Starting with 3.7, maintaining dictionary order is part of the language. And as an implementation detail CPython has done this since 3.5. I think we should be okay, so long as we are not directly using the hash. We could also run the tests with a fixed value for PYTHONHASHSEED, if needed. Or am I not understanding the issue here correctly?

Reply:
Agreed that this won't be a problem in Python 3, but until we move over unpredictable iterations are a problem. Sorting the dictionary could be an alternative to making a lot of other changes to the test.

But Jonathan already made those changes so it's not a big deal given the short lifespan of this code.